<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#General-Information" data-toc-modified-id="General-Information-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>General Information</a></span><ul class="toc-item"><li><span><a href="#What-are-open-access-datasets?" data-toc-modified-id="What-are-open-access-datasets?-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>What are open access datasets?</a></span></li><li><span><a href="#Why-would-I-want-to-use-an-open-access-dataset?" data-toc-modified-id="Why-would-I-want-to-use-an-open-access-dataset?-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Why would I want to use an open access dataset?</a></span></li><li><span><a href="#Where-can-I-find-them?" data-toc-modified-id="Where-can-I-find-them?-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Where can I find them?</a></span></li><li><span><a href="#Gaining-access" data-toc-modified-id="Gaining-access-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Gaining access</a></span></li><li><span><a href="#Logistical-issues" data-toc-modified-id="Logistical-issues-1.5"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Logistical issues</a></span><ul class="toc-item"><li><span><a href="#Hosting-the-data" data-toc-modified-id="Hosting-the-data-1.5.1"><span class="toc-item-num">1.5.1&nbsp;&nbsp;</span>Hosting the data</a></span></li><li><span><a href="#Learning-the-data" data-toc-modified-id="Learning-the-data-1.5.2"><span class="toc-item-num">1.5.2&nbsp;&nbsp;</span>Learning the data</a></span></li></ul></li></ul></li><li><span><a href="#HCP-Young-Adult-Behavioral-Data" data-toc-modified-id="HCP-Young-Adult-Behavioral-Data-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>HCP Young Adult Behavioral Data</a></span><ul class="toc-item"><li><span><a href="#Data-Dictionary" data-toc-modified-id="Data-Dictionary-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Data Dictionary</a></span><ul class="toc-item"><li><span><a href="#Import-data-dictionary" data-toc-modified-id="Import-data-dictionary-2.1.1"><span class="toc-item-num">2.1.1&nbsp;&nbsp;</span>Import data dictionary</a></span></li><li><span><a href="#Search-for-a-specific-variable." data-toc-modified-id="Search-for-a-specific-variable.-2.1.2"><span class="toc-item-num">2.1.2&nbsp;&nbsp;</span>Search for a specific variable.</a></span></li><li><span><a href="#Filtering-a-dataframe" data-toc-modified-id="Filtering-a-dataframe-2.1.3"><span class="toc-item-num">2.1.3&nbsp;&nbsp;</span>Filtering a dataframe</a></span></li><li><span><a href="#Combine-dataframes" data-toc-modified-id="Combine-dataframes-2.1.4"><span class="toc-item-num">2.1.4&nbsp;&nbsp;</span>Combine dataframes</a></span></li><li><span><a href="#Run-Stats" data-toc-modified-id="Run-Stats-2.1.5"><span class="toc-item-num">2.1.5&nbsp;&nbsp;</span>Run Stats</a></span></li></ul></li></ul></li><li><span><a href="#HCP-Freesurfer-Structural-Data" data-toc-modified-id="HCP-Freesurfer-Structural-Data-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>HCP Freesurfer Structural Data</a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Filter-dataframe" data-toc-modified-id="Filter-dataframe-3.0.1"><span class="toc-item-num">3.0.1&nbsp;&nbsp;</span>Filter dataframe</a></span></li><li><span><a href="#Combine-behavioral-and-neural-dataframes" data-toc-modified-id="Combine-behavioral-and-neural-dataframes-3.0.2"><span class="toc-item-num">3.0.2&nbsp;&nbsp;</span>Combine behavioral and neural dataframes</a></span></li><li><span><a href="#Run-Stats" data-toc-modified-id="Run-Stats-3.0.3"><span class="toc-item-num">3.0.3&nbsp;&nbsp;</span>Run Stats</a></span></li></ul></li></ul></li></ul></div>

# General Information

## What are open access datasets?
Open access datasets are datasets that are free and accessible to "anyone". These datasets have been collected by a group of researchers who have decided to share their data with the scientific community in order to promote openness, reproducibility, and more research by allowing their datasets to be reused instead of being locked away on some hard drive. Typically, one must register to a website which hosts the data and create an account where they agree to the dataset's term of use. There may be an approval process that you need to go through before you receive access. <br>

There are two types of open access datasets: 

* __Datasets that are from a single group/study.__ These are studies that are just like any other a lab would run. The researchers finish their typical data collection and upload the data to a repository after their paper is published.

* __Large multi-site projects.__ These are datasets that come from multiple labs/researchers that are combined into a single dataset and hosted together. Some of these datasets are combined "post-hoc", and were not originally collected with the goal being to combine them into a single dataset. Recently, there have been efforts to fully standardize data collection at the multiple sites with the intention of creating an open access dataset. These datasets are usually of higher quality.

## Why would I want to use an open access dataset?
* Pros
    * Large sample sizes - Open access datasets typically have very large sample sizes, as they are often multi-site studies.
    * Free data collection
    * Data can be very high quality, which many researchers might not be able to acquire themselves
    * Data can come preprocessed
    * Data (especially behavioral data) is usually clean
* Cons
    * Might not be able to test your specific hypotheses
    * Some multi-side studies can be very troublesome
    * Data can come preprocessed
    * [Dataset decay](https://elifesciences.org/articles/53498) - overusing open access datasets may increase error rates


## Where can I find them?
* [OpenNeuro](https://openneuro.org/)
    * Database of publically available single-study datasets
* [Brain Genomics Superstruct Project (GSP)](https://dataverse.harvard.edu/dataverse/GSP)<img src=images/gsp.jpg />
    * A collection of neuroimaing, behavior, cognitive, and personality data for over 1,500 human participants
    * This dataset was used in [Yeo et al., 2011](https://journals.physiology.org/doi/full/10.1152/jn.00338.2011) to create the widely used Yeo 7-network atlas
* [Adolescent Brain Cognitive Development (ABCD)](https://nda.nih.gov/abcd)
    * Longitudinal study of brain development and child health from 21 sites
    * Measured social, emotional, cognitive development, as well as health and environmental outcomes
    * This dataset is somewhat notorious for its various issues
        * The imaging data is very troublesome, as it is from different sites, with different scanners, scanner software, head-coils, etc.
* [Human Connectome Project (HCP)](https://www.humanconnectome.org/study/hcp-young-adult)<img src=images/hcp_mmp.PNG />
    * The HCP young adult dataset was collected on the same MRI scanner model across three different sites
    * It is very high quality considering the multi-site nature of the dataset and the large sample size (1200)
        * The researchers took careful consideration of the issues with multi-site studies and ensured that all studies were using the same type of scanner, scanner software, head-coil, and protocol
    * Includes multi-modal data from fMRI tasks, fMRI resting state, DWI, and high resolution T1 structural scans, along with behavioral and cognitive data
    * Most of the data is available fully preprocessed via their [Minimal Preprocessing Pipeline](https://pubmed.ncbi.nlm.nih.gov/23668970/)
    * Additional HCP datasets
        * [HCP Development](https://www.humanconnectome.org/study/hcp-lifespan-development)
        * [HCP Aging](https://www.humanconnectome.org/study/hcp-lifespan-aging)
        * Human disease (Alzheimer's, anxiety and depression, etc.) datasets to come
    
* [Autism Brain Imaging Data Exchange (ABIDE)](http://fcon_1000.projects.nitrc.org/indi/abide/abide_I.html)
* [Open Access Series of Imaging Studies (OASIS)](https://www.oasis-brains.org/)
* [Alzheimer's Disease Neuroimaing Initiative (ADNI)](http://adni.loni.usc.edu/)



## Gaining access
It turns out that "open access" is not as accessible as you would think. Typically, researchers will have to apply to gain access to a dataset. This can be done by signing up and creating an account for the dataset repository, or submitting an pseudo application in which you will have to say who you are and what you intend to use the data for. <br>

For this workshop, we will focus on the young adult (original) HCP dataset. The following steps will help you gain access to this dataset:

1. Create an account on [ConnectomeDB](https://db.humanconnectome.org/app/template/Login.vm;jsessionid=5629BEC4DD00C13DB36AAD1C07677226).
<br>

2. Login with your connectomeDB account
<br>

3. Install the Aspera Connect plugin <img src=https://raw.githubusercontent.com/TU-Coding-Outreach-Group/cog_summer_workshops_2020/master/open-access-datasets/images/aspera_connect.PNG height=100 width=700 />
<br>

4. Log back into [ConnectomeDB](https://db.humanconnectome.org/app/template/Login.vm;jsessionid=5629BEC4DD00C13DB36AAD1C07677226) and accept the usage terms.
<br>

5. Download the behavioral and Freesurfer structural data <img height=400 width=500 src=https://raw.githubusercontent.com/TU-Coding-Outreach-Group/cog_summer_workshops_2020/master/open-access-datasets/images/hcp_download.PNG />
<br>




## Logistical issues

### Hosting the data
Some datasets, like neuroimaging data, can be very, VERY large. For example, a single subject's MRI data from the young adult HCP is over 40GB. Multiple that by 1200 (or slighly less) subjects, and you get 48TB of data, which is impossible for a single computer to host. In addition, it would be inefficient to have multiple researchers in a lab all have the same (large) dataset on their personal computers. Instead, there are two options: 1) Find a server to host the data for you in a common space that is accessible to everyone, or 2) only download a subset of the data. For the young adult HCP data, our lab has gone with option 1 and has teamed up with the computer science high performance cluster people at Temple to host the data on Owlsnest. 

### Learning the data
Open access datasets may have variables that you are unfamiliar with, or that are named/abbreviated a specific way. To find out what variables are and what they aim to measure, refer to the dataset's data dictionary. Datasets may also come with documentation that describe what the dataset is, how it was acquired, how the data is structed and named. For imaging datasets, this is particularly important as the documentation will help you navigate directory structures to find the data you are interested in.


***
# HCP Young Adult Behavioral Data
For this workshop, we will explore the behavioral and Freesurfer structural data from the young adult HCP data. Once these have been downloaded, place them somewhere where you can access them via the command line. The data was not provided in this repo, as that may be breaking the dataset's terms of use.

In [7]:
library(dplyr)

Warning message:
"package 'dplyr' was built under R version 3.6.2"
Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union



## Data Dictionary
When we open the  behavioral data spreadsheet, we can see that variables are named as acronyms that might be unfamiliar. A data dictionary is a file (usually a csv or excel file) that acts as a sort of key to tell you what variables are, how they are named, and what they measure. <br>

Below, we can explore some of the variables in the behavioral data by looking at the column names.

In [108]:
behav_data <- read.csv('/Users/Administrator/Downloads/unrestricted_hspopal_6_17_2020_13_26_49.csv')
head(behav_data)

Subject Release Acquisition Gender Age   X3T_Full_MR_Compl T1_Count T2_Count
1 100004  S900    Q06         M      22-25 false             0        0       
2 100206  S900    Q11         M      26-30 true              1        1       
3 100307  Q1      Q01         F      26-30 true              1        1       
4 100408  Q3      Q03         M      31-35 true              1        1       
5 100610  S900    Q08         M      26-30 true              2        1       
6 101006  S500    Q06         F      31-35 true              2        2       
  X3T_RS.fMRI_Count X3T_RS.fMRI_PctCompl ... Noise_Comp Odor_Unadj Odor_AgeAdj
1 0                   0                  ... 5.2        101.12      86.45     
2 4                 100                  ... 6.0        108.79      97.19     
3 4                 100                  ... 3.6        101.12      86.45     
4 4                 100                  ... 2.0        108.79      98.04     
5 4                 100                  ... 2.0        122.25     110.45     
6 4                 100                  ... 6.0        122.25     111.41     
  PainIntens_RawScore PainInterf_Tscore Taste_Unadj Taste_AgeAdj Mars_Log_Score
1 2                   45.9              107.17      105.31       1.80          
2 1                   49.7               72.63       72.03       1.84          
3 0                   38.6               71.69       71.76       1.76          
4 2                   52.6              114.01      113.59       1.76          
5 0                   38.6               84.84       85.31       1.92          
6 0                   38.6              123.80      123.31       1.80          
  Mars_Errs Mars_Final
1 0         1.80      
2 0         1.84      
3 0         1.76      
4 2         1.68      
5 1         1.88      
6 0         1.80

### Import data dictionary
To find out what these variables are, we need to refer to the data dictionary for the dataset. Below, the dictionary is important and we can explore it a little. 

In [109]:
data_dict <- read.csv('/Users/Administrator/Downloads/HCP_S1200_DataDictionary_April_20_2018.csv')
head(data_dict)

fullDisplayName     category            assessment   columnHeader
1 Subject             Subject Information Demographics Subject     
2 Quarter Released    Subject Information Demographics Release     
3 Acquisition Quarter Subject Information Demographics Acquisition 
4 Gender              Subject Information Demographics Gender      
5 Age Range           Subject Information Demographics Age         
6 Age Range           Subject Information Demographics Age         
  description                                                                                                                                                                                                                                                                            
1 HCP Subject ID                                                                                                                                                                                                                                                                         
2 HCP data release in which this subject's data was initially published to ConnectomeDB.   Note: 'Release' does not indicate the version of the pipelines used to process subject data. Check the release-notes distributed with each processed dataset for pipeline version information.
3 Quarter in which this subject's data was initially acquired.                                                                                                                                                                                                                           
4 Gender of Subject                                                                                                                                                                                                                                                                      
5 Age group of Participant, banded in five-year increments.                                                                                                                                                                                                                              
6 Age group of Participant.

Two important things to note. The "columnHeader" column contains the list of variables found in the behavioral data, and the "description" column gives some information that variable and what particular domain or question it is related to.

### Search for a specific variable.
We can use this bit of code below to find revelant information about a specific variable in the data dictionary.

In [112]:
data_dict[data_dict$columnHeader == 'PSQI_Bathroom',]

fullDisplayName                            category 
178 5c. Sleep Trouble - Get up to use bathroom Alertness
    assessment                             columnHeader 
178 Sleep (Pittsburgh Sleep Questionnaire) PSQI_Bathroom
    description                                                                                                                                                                                                                         
178 PSQI 5.During the past month, how often have you had trouble sleeping because you...  (c) Have to get up to use the bathroom; 0=Not during the past month, 1=Less than once a week, 2=Once or twice a week, 3=3 or more times a week

If we want to look up the information for all of the variables from the PSQI, we can use the bit of code below to filter the data dictionary to only present the rows that have the string "PSQI" in the columnHeader column.

In [113]:
filter(data_dict, grepl('PSQI', columnHeader))

fullDisplayName                                                
1  Pittsburgh Sleep Quality Index (PSQI) Completed                
2  Sleep (Pittsburgh Sleep Questionnaire) Total Score             
3  Sleep (Pittsburgh Sleep Questionnaire) Component 1 Score       
4  Sleep (Pittsburgh Sleep Questionnaire) Component 2 Score       
5  Sleep (Pittsburgh Sleep Questionnaire) Component 3 Score       
6  Sleep (Pittsburgh Sleep Questionnaire) Component 4 Score       
7  Sleep (Pittsburgh Sleep Questionnaire) Component 5 Score       
8  Sleep (Pittsburgh Sleep Questionnaire) Component 6 Score       
9  Sleep (Pittsburgh Sleep Questionnaire) Component 7 Score       
10 1. Usual bed time (past month)                                 
11 2. Minutes to fall asleep (past month)                         
12 3. Time get up in morning (past month)                         
13 4. Hours of sleep per night (past month)                       
14 5a. Sleep Trouble - Can't get to sleep within 30 minutes       
15 5b. Sleep Trouble - Wake up in middle of night or early morning
16 5c. Sleep Trouble - Get up to use bathroom                     
17 5d. Sleep Trouble - Can't breathe comfortably                  
18 5e. Sleep Trouble - Cough or snore loudly                      
19 5f. Sleep Trouble - Feel too cold                              
20 5g. Sleep Trouble - Feel too hot                               
21 5h. Sleep Trouble - Had bad dreams                             
22 5i. Sleep Trouble - Have pain                                  
23 5j. Sleep Trouble - Other                                      
24 6. Describe overall sleep quality                              
25 7. How often taken sleep medicine                              
26 8. How often trouble staying awake                             
27 9. How often trouble keeping up enthusiasm                     
28 10. Have bed partner or roommate                               
   category         assessment                             columnHeader      
1  Study Completion Study Completion: Behavior             PSQI_Compl        
2  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Score        
3  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp1        
4  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp2        
5  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp3        
6  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp4        
7  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp5        
8  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp6        
9  Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Comp7        
10 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_BedTime      
11 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Min2Asleep   
12 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_GetUpTime    
13 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_AmtSleep     
14 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Latency30Min 
15 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_WakeUp       
16 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Bathroom     
17 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Breathe      
18 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Snore        
19 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_TooCold      
20 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_TooHot       
21 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_BadDream     
22 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Pain         
23 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Other        
24 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_Quality      
25 Alertness        Sleep (Pittsburgh Sleep Questionnaire) PSQI_SleepMeds    
26 Alertness        Sleep (Pitts

### Filtering a dataframe
Next, we will filter the behavioral data and create a new dataframe that only has the PSQI variables in it.

In [57]:
psqi_vars <- filter(data_dict, grepl('PSQI', columnHeader))
psqi_vars <- psqi_vars$columnHeader
psqi_data <- select(filter(behav_data),psqi_vars)
psqi_data$Subject <- behav_data$Subject

head(psqi_data)
nrow(psqi_data)

PSQI_Compl PSQI_Score PSQI_Comp1 PSQI_Comp2 PSQI_Comp3 PSQI_Comp4 PSQI_Comp5
1 true       8          1          2          2          1          1         
2 true       6          1          1          1          1          1         
3 true       4          1          0          1          0          2         
4 true       4          1          1          0          0          1         
5 true       4          1          1          0          0          1         
6 true       2          1          1          0          0          0         
  PSQI_Comp6 PSQI_Comp7 PSQI_BedTime ... PSQI_TooHot PSQI_BadDream PSQI_Pain
1 0          1          09:00:00     ... 1           0             0        
2 0          1          22:30:00     ... 1           0             0        
3 0          0          22:00:00     ... 1           0             0        
4 0          1          22:00:00     ... 0           0             0        
5 0          1          21:30:00     ... 1           0             0        
6 0          0          23:00:00     ... 0           0             0        
  PSQI_Other PSQI_Quality PSQI_SleepMeds PSQI_DayStayAwake PSQI_DayEnthusiasm
1 0          1            0              0                 2                 
2 1          1            0              1                 1                 
3 3          1            0              0                 0                 
4 0          1            0              0                 1                 
5 0          1            0              0                 1                 
6 0          1            0              0                 0                 
  PSQI_BedPtnrRmate Subject
1 0                 100004 
2 0                 100206 
3 3                 100307 
4 3                 100408 
5 0                 100610 
6 3                 101006

[1] 1206

The last command in the code above gives the number of rows (or observations) in the dataframe. This tell us that we have data for 1206 subjects. <br>

Now, we will add a couple of extra variables to this dataframe. We will pull age and a variable that measures the subject's level of angry affect. These two variables will have to be added to a new dataframe, before they can be combined with the PSQI data.

In [30]:
add_vars <- c('Age','AngAffect_Unadj')
add_data <- select(filter(behav_data),add_vars)
head(add_data)

Note: Using an external vector in selections is ambiguous.
i Use `all_of(add_vars)` instead of `add_vars` to silence this message.
i See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.
This message is displayed once per session.


Age   AngAffect_Unadj
1 22-25 49.3           
2 26-30 54.8           
3 26-30 46.9           
4 31-35 50.6           
5 26-30 47.5           
6 31-35 59.0

### Combine dataframes
Now we will combine the dataframe that contains the age and angry affect data with the data frame that contains the PSQI data.

In [114]:
psqi_data <- cbind(add_data, psqi_data)
summary(psqi_data)

    Age      AngAffect_Unadj PSQI_Compl    PSQI_Score       PSQI_Comp1    
 22-25:247   Min.   :28.60   true:1206   Min.   : 0.000   Min.   :0.0000  
 26-30:527   1st Qu.:43.00               1st Qu.: 3.000   1st Qu.:1.0000  
 31-35:418   Median :48.40               Median : 4.000   Median :1.0000  
 36+  : 14   Mean   :48.03               Mean   : 4.855   Mean   :0.9038  
             3rd Qu.:53.10               3rd Qu.: 6.000   3rd Qu.:1.0000  
             Max.   :85.40               Max.   :19.000   Max.   :3.0000  
             NA's   :1                                                    
   PSQI_Comp2       PSQI_Comp3       PSQI_Comp4      PSQI_Comp5   
 Min.   :0.0000   Min.   :0.0000   Min.   :0.000   Min.   :0.000  
 1st Qu.:0.0000   1st Qu.:0.0000   1st Qu.:0.000   1st Qu.:1.000  
 Median :1.0000   Median :0.0000   Median :0.000   Median :1.000  
 Mean   :0.9818   Mean   :0.5755   Mean   :0.466   Mean   :1.105  
 3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:1.000   3rd Qu.:1.000 

### Run Stats
Below, we will run a simple Pearson correlation.

In [115]:
corr_results <- cor.test(relv_data$AngAffect_Unadj, relv_data$PSQI_Bathroom)
corr_results


	Pearson's product-moment correlation

data:  relv_data$AngAffect_Unadj and relv_data$PSQI_Bathroom
t = 4.137, df = 1203, p-value = 3.763e-05
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.06238156 0.17374635
sample estimates:
      cor 
0.1184364 


For accessibility for easier reporting when I write my results section, I like to report my stats as I would in a paper.

In [116]:
print(paste0("Having trouble sleeping because of a need to use the bathroom is weakly correlated with angry affect (r = ", 
             round(corr_results$estimate, digits = 4),
            ", p = ",corr_results$p.value,")"))

[1] "Having trouble sleeping because of a need to use the bathroom is weakly correlated with angry affect (r = 0.1184, p = 3.76344677497135e-05)"


***
# HCP Freesurfer Structural Data
Next, we will explore the HCP Freesurfer data. Freesurfer is a neuroimaging program that specializes in structural analyses. The Freesurfer reconstruction pipeline takes in a standard T1 image and after about 12 hours of preprocessing, exports brain data related to various measures, tissues, parcellations, and stats. The benefit of the HCP data is that all of this work is done for you. If you wanted to do whole brain analyses, you could download the Freesurfer data for any subject and complete your own analyses. They HCP folks have also aggregated data on brain measures related to structural measures such as cortical thickness, surface area, and volume for all of the subjects, into a single spreadsheet. We will explore this data below.

In [1]:
struct_data <- read.csv('/Users/Administrator/Downloads/unrestricted_hcp_freesurfer.csv')

head(struct_data)

Subject Gender Release FS_InterCranial_Vol FS_BrainSeg_Vol
1 100206  M      S900    1864518             1406924        
2 100307  F      Q1      1512540             1109710        
3 100408  M      Q3      1684117             1280171        
4 100610  M      S900    1816044             1321362        
5 101006  F      S500    1437868             1053766        
6 101107  M      S500    1512727             1126655        
  FS_BrainSeg_Vol_No_Vent FS_BrainSeg_Vol_No_Vent_Surf FS_LCort_GM_Vol
1 1389007                 1386836                      304488         
2 1098854                 1097122                      240030         
3 1260943                 1259250                      260695         
4 1306929                 1304569                      276506         
5 1038408                 1037793                      203718         
6 1111763                 1109747                      242489         
  FS_RCort_GM_Vol FS_TotCort_GM_Vol ... FS_R_Rostralanteriorcingulate_CurvInd
1 305634          610122            ... 5.6                                  
2 243207          483238            ... 3.3                                  
3 265243          525938            ... 3.3                                  
4 278002          554508            ... 3.0                                  
5 210539          414257            ... 2.9                                  
6 248807          491296            ... 3.9                                  
  FS_R_Rostralmiddlefrontal_CurvInd FS_R_Superiorfrontal_CurvInd
1 34.6                              40.1                        
2 29.9                              48.8                        
3 25.2                              27.5                        
4 19.6                              20.0                        
5 14.7                              21.3                        
6 17.5                              18.1                        
  FS_R_Superiorparietal_CurvInd FS_R_Superiortemporal_CurvInd
1 23.3                          15.6                         
2 21.6                          18.6                         
3 15.2                          13.9                         
4 18.6                          10.6                         
5 13.2                           8.7                         
6  9.9                           6.7                         
  FS_R_Supramarginal_CurvInd FS_R_Frontalpole_CurvInd FS_R_Temporalpole_CurvInd
1 11.9                       2.9                      3.8                      
2 15.0                       2.9                      5.8                      
3 12.3                       1.8                      7.8                      
4 10.8                       1.2                      4.1                      
5  9.8                       1.1                      3.0                      
6  7.2                       1.1                      2.3                      
  FS_R_Transversetemporal_CurvInd FS_R_Insula_CurvInd
1 2.8                             37.5               
2 2.0                             30.2               
3 1.9                             25.1               
4 1.4                             16.6               
5 1.0                             18.9               
6 0.9                             12.1

The dataset has brain measures for specific brain regions from the [Desikan Atlas](https://surfer.nmr.mgh.harvard.edu/fswiki/CorticalParcellation). For each of these brain regions, we can find values for measures like cortical thickness, surface area, and volume.

### Filter dataframe
We will filter this dataset to only include data on cortical thickness. We can see that we have 1113 observations, so some subjects do not have Freesurfer structural data.

In [55]:
thick_data <- select(struct_data, ends_with('Thck'))
thick_data$Subject <- struct_data$Subject

head(thick_data)
nrow(thick_data)

FS_L_Bankssts_Thck FS_L_Caudalanteriorcingulate_Thck
1 2.733              2.682                            
2 2.713              3.253                            
3 2.532              2.588                            
4 2.683              3.029                            
5 2.632              2.657                            
6 2.877              2.729                            
  FS_L_Caudalmiddlefrontal_Thck FS_L_Cuneus_Thck FS_L_Entorhinal_Thck
1 2.765                         2.185            3.388               
2 2.795                         1.957            3.389               
3 2.612                         2.127            3.310               
4 2.735                         2.223            3.664               
5 2.839                         1.977            3.236               
6 2.851                         2.251            3.529               
  FS_L_Fusiform_Thck FS_L_Inferiorparietal_Thck FS_L_Inferiortemporal_Thck
1 2.957              2.583                      3.065                     
2 3.062              2.637                      2.968                     
3 2.870              2.515                      2.840                     
4 2.873              2.547                      3.062                     
5 2.841              2.522                      2.846                     
6 3.041              2.769                      3.123                     
  FS_L_Isthmuscingulate_Thck FS_L_Lateraloccipital_Thck ...
1 2.428                      2.304                      ...
2 2.533                      2.259                      ...
3 2.257                      2.247                      ...
4 2.471                      2.238                      ...
5 2.300                      2.231                      ...
6 2.657                      2.529                      ...
  FS_R_Rostralmiddlefrontal_Thck FS_R_Superiorfrontal_Thck
1 2.515                          2.731                    
2 2.789                          3.087                    
3 2.502                          2.760                    
4 2.620                          3.013                    
5 2.581                          2.864                    
6 2.727                          3.038                    
  FS_R_Superiorparietal_Thck FS_R_Superiortemporal_Thck FS_R_Supramarginal_Thck
1 2.278                      3.032                      2.660                  
2 2.250                      3.151                      2.766                  
3 2.282                      2.846                      2.554                  
4 2.257                      2.901                      2.632                  
5 2.313                      2.879                      2.674                  
6 2.502                      3.151                      2.895                  
  FS_R_Frontalpole_Thck FS_R_Temporalpole_Thck FS_R_Transversetemporal_Thck
1 2.641                 3.579                  3.147                       
2 3.675                 4.026                  2.819                       
3 2.840                 3.102                  2.532                       
4 2.994                 3.723                  2.721                       
5 2.658                 3.540                  2.726                       
6 2.976                 3.956                  2.649                       
  FS_R_Insula_Thck Subject
1 3.278            100206 
2 3.002            100307 
3 2.947            100408 
4 3.068            100610 
5 2.958            101006 
6 3.346            101107

[1] 1113

### Combine behavioral and neural dataframes
Now, we will combine the cortical thickness data with the PSQI, by matching the subject IDs between the two datasets. This is very important, so that a subject's PSQI data is directly matched with their brain data.

In [54]:
psqi_thick_data <- merge(psqi_data, thick_data, by='Subject')

head(psqi_thick_data)
nrow(psqi_thick_data)

Subject PSQI_Compl PSQI_Score PSQI_Comp1 PSQI_Comp2 PSQI_Comp3 PSQI_Comp4
1 100206  true       6          1          1          1          1         
2 100307  true       4          1          0          1          0         
3 100408  true       4          1          1          0          0         
4 100610  true       4          1          1          0          0         
5 101006  true       2          1          1          0          0         
6 101107  true       9          1          1          3          3         
  PSQI_Comp5 PSQI_Comp6 PSQI_Comp7 ... FS_R_Rostralanteriorcingulate_Thck
1 1          0          1          ... 3.115                             
2 2          0          0          ... 3.012                             
3 1          0          1          ... 2.862                             
4 1          0          1          ... 3.336                             
5 0          0          0          ... 2.807                             
6 0          0          1          ... 3.278                             
  FS_R_Rostralmiddlefrontal_Thck FS_R_Superiorfrontal_Thck
1 2.515                          2.731                    
2 2.789                          3.087                    
3 2.502                          2.760                    
4 2.620                          3.013                    
5 2.581                          2.864                    
6 2.727                          3.038                    
  FS_R_Superiorparietal_Thck FS_R_Superiortemporal_Thck FS_R_Supramarginal_Thck
1 2.278                      3.032                      2.660                  
2 2.250                      3.151                      2.766                  
3 2.282                      2.846                      2.554                  
4 2.257                      2.901                      2.632                  
5 2.313                      2.879                      2.674                  
6 2.502                      3.151                      2.895                  
  FS_R_Frontalpole_Thck FS_R_Temporalpole_Thck FS_R_Transversetemporal_Thck
1 2.641                 3.579                  3.147                       
2 3.675                 4.026                  2.819                       
3 2.840                 3.102                  2.532                       
4 2.994                 3.723                  2.721                       
5 2.658                 3.540                  2.726                       
6 2.976                 3.956                  2.649                       
  FS_R_Insula_Thck
1 3.278           
2 3.002           
3 2.947           
4 3.068           
5 2.958           
6 3.346

[1] 1113

### Run Stats
Below is a list of all of the brain regions.

In [53]:
colnames(thick_data)[1:68]

[1] "FS_L_Bankssts_Thck"                 "FS_L_Caudalanteriorcingulate_Thck" 
 [3] "FS_L_Caudalmiddlefrontal_Thck"      "FS_L_Cuneus_Thck"                  
 [5] "FS_L_Entorhinal_Thck"               "FS_L_Fusiform_Thck"                
 [7] "FS_L_Inferiorparietal_Thck"         "FS_L_Inferiortemporal_Thck"        
 [9] "FS_L_Isthmuscingulate_Thck"         "FS_L_Lateraloccipital_Thck"        
[11] "FS_L_Lateralorbitofrontal_Thck"     "FS_L_Lingual_Thck"                 
[13] "FS_L_Medialorbitofrontal_Thck"      "FS_L_Middletemporal_Thck"          
[15] "FS_L_Parahippocampal_Thck"          "FS_L_Paracentral_Thck"             
[17] "FS_L_Parsopercularis_Thck"          "FS_L_Parsorbitalis_Thck"           
[19] "FS_L_Parstriangularis_Thck"         "FS_L_Pericalcarine_Thck"           
[21] "FS_L_Postcentral_Thck"              "FS_L_Posteriorcingulate_Thck"      
[23] "FS_L_Precentral_Thck"               "FS_L_Precuneus_Thck"               
[25] "FS_L_Rostralanteriorcingulate_Thck" "FS_L_Rostralmiddlefrontal_Thck"    
[27] "FS_L_Superiorfrontal_Thck"          "FS_L_Superiorparietal_Thck"        
[29] "FS_L_Superiortemporal_Thck"         "FS_L_Supramarginal_Thck"           
[31] "FS_L_Frontalpole_Thck"              "FS_L_Temporalpole_Thck"            
[33] "FS_L_Transversetemporal_Thck"       "FS_L_Insula_Thck"                  
[35] "FS_R_Bankssts_Thck"                 "FS_R_Caudalanteriorcingulate_Thck" 
[37] "FS_R_Caudalmiddlefrontal_Thck"      "FS_R_Cuneus_Thck"                  
[39] "FS_R_Entorhinal_Thck"               "FS_R_Fusiform_Thck"                
[41] "FS_R_Inferiorparietal_Thck"         "FS_R_Inferiortemporal_Thck"        
[43] "FS_R_Isthmuscingulate_Thck"         "FS_R_Lateraloccipital_Thck"        
[45] "FS_R_Lateralorbitofrontal_Thck"     "FS_R_Lingual_Thck"                 
[47] "FS_R_Medialorbitofrontal_Thck"      "FS_R_Middletemporal_Thck"          
[49] "FS_R_Parahippocampal_Thck"          "FS_R_Paracentral_Thck"             
[51] "FS_R_Parsopercularis_Thck"          "FS_R_Parsorbitalis_Thck"           
[53] "FS_R_Parstriangularis_Thck"         "FS_R_Pericalcarine_Thck"           
[55] "FS_R_Postcentral_Thck"              "FS_R_Posteriorcingulate_Thck"      
[57] "FS_R_Precentral_Thck"               "FS_R_Precuneus_Thck"               
[59] "FS_R_Rostralanteriorcingulate_Thck" "FS_R_Rostralmiddlefrontal_Thck"    
[61] "FS_R_Superiorfrontal_Thck"          "FS_R_Superiorparietal_Thck"        
[63] "FS_R_Superiortemporal_Thck"         "FS_R_Supramarginal_Thck"           
[65] "FS_R_Frontalpole_Thck"              "FS_R_Temporalpole_Thck"            
[67] "FS_R_Transversetemporal_Thck"       "FS_R_Insula_Thck"

We will see if the measure of problems sleeping because of a need to use the bathroom is correlated to cortical thickness in any of the brain regions from the Desikan atlas. We will do this by looping through each brain region variable and correlating it to the sleep measure.

In [101]:
corr_results <- data.frame(region=character(0),
                          r=numeric(0),
                          p=numeric(0))

for (roi in colnames(thick_data)[1:68]){
    temp_result <- cor.test(psqi_thick_data$PSQI_Bathroom, psqi_thick_data[[roi]])
    temp_df <- data.frame(region=roi, r=temp_result$estimate, p=temp_result$p.value)
    corr_results <- rbind(corr_results, temp_df)
}

head(corr_results)

region                            r           p          
cor  FS_L_Bankssts_Thck                -0.08162860 0.006434957
cor1 FS_L_Caudalanteriorcingulate_Thck -0.02065750 0.491158403
cor2 FS_L_Caudalmiddlefrontal_Thck     -0.02385389 0.426599399
cor3 FS_L_Cuneus_Thck                  -0.05221451 0.081649794
cor4 FS_L_Entorhinal_Thck              -0.01471964 0.623747342
cor5 FS_L_Fusiform_Thck                -0.05317973 0.076158288

We can see that there are many brain regions related to this sleep measure. 

In [106]:
corr_results[corr_results$p < 0.05,]

region                         r           p           
cor   FS_L_Bankssts_Thck             -0.08162860 0.0064349566
cor9  FS_L_Lateraloccipital_Thck     -0.06113074 0.0414462250
cor10 FS_L_Lateralorbitofrontal_Thck -0.07554982 0.0116942501
cor18 FS_L_Parstriangularis_Thck     -0.06399382 0.0327831456
cor28 FS_L_Superiortemporal_Thck     -0.07267394 0.0153081437
cor34 FS_R_Bankssts_Thck             -0.09007043 0.0026328733
cor37 FS_R_Cuneus_Thck               -0.06677933 0.0258903259
cor38 FS_R_Entorhinal_Thck           -0.07100491 0.0178280455
cor39 FS_R_Fusiform_Thck             -0.07658352 0.0105933004
cor40 FS_R_Inferiorparietal_Thck     -0.06222014 0.0379452758
cor41 FS_R_Inferiortemporal_Thck     -0.10475286 0.0004644977
cor43 FS_R_Lateraloccipital_Thck     -0.09284754 0.0019302022
cor44 FS_R_Lateralorbitofrontal_Thck -0.08099766 0.0068588140
cor51 FS_R_Parsorbitalis_Thck        -0.05927398 0.0480422445
cor62 FS_R_Superiortemporal_Thck     -0.08666303 0.0038107330
cor63 FS_R_Supramarginal_Thck        -0.07120410 0.0175093630
cor65 FS_R_Temporalpole_Thck         -0.09441405 0.0016142714
cor66 FS_R_Transversetemporal_Thck   -0.09261462 0.0019817448
cor67 FS_R_Insula_Thck               -0.07441763 0.0130153329

However, since we ran many correlations, we should correct for multiple comparisons. This is something that one should pay careful attention to with large datasets as there are many variables and it is easy to find any kind of relationship between the variables. <br>

Below, we will calculate a new p-value based on a Bonferroni correction. Then we will show the brain regions that are now significantly related to problems sleeping because of a need to use the bathroom.

In [107]:
bonf_p = 0.05/68

corr_results[corr_results$p < bonf_p,]

region                     r          p           
cor41 FS_R_Inferiortemporal_Thck -0.1047529 0.0004644977